# Project: Profitable App Profiles for the App Store and Google Play Market

# I. Introduction
## 1. Intoduce Project
Our aim in this project is to find mobile app profiles that are profitable for the App Store and Google Play markets.

We have the standards:

1. **Global**
> this means that the applications are easily approach to everyone and without difficulty about price, language,...

2. **Profitable**
> because our revenue is highly influenced by the number of people using our apps.

## 2. About dataset
As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play.

![](https://s3.amazonaws.com/dq-content/350/py1m8_statista.png)

> Link download dataset:
> * [**Google Play Store Apps**](https://www.kaggle.com/lava18/google-play-store-apps)
> * [**Mobile App Store**](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps)

# II. Explore Dataset

## 1. Load data set

In [1]:
from csv import reader
def load_data(dataset):
    open_file = open(dataset, encoding='utf8')
    read_file = reader(open_file)
    list_app = list(read_file)
    app_header = list_app[0]
    list_app = list_app[1:]
    return app_header, list_app

In [2]:
android_header, android = load_data('googleplaystore.csv')

In [3]:
ios_header, ios = load_data('AppleStore.csv')

## 2. Explore Dataset

In [4]:
def explore_data(dataset, start = 0, end = 5, rows_and_columns = True):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
    if rows_and_columns:
        print('Number of rows:', len(dataset))

### 2.1. Google Playstore

In [5]:
print("Header of android:\n",android_header)
print("length header: ", len(android_header))

Header of android:
 ['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
length header:  13


In [6]:
explore_data(android, 0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']
['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']
['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']
Number of rows: 10841


### 2.2. Apple Store

In [7]:
print("Ios of android:\n",ios_header)
print("length header: ", len(ios_header))

Ios of android:
 ['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
length header:  16


In [8]:
explore_data(ios, 0,3,True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']
['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']
['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']
Number of rows: 7197


# III. Cleaning Data
*We need to make sure the data we analyze is accurate, otherwise the results of our analysis will be wrong.* 

**We need to:**
- Detect inaccurate data, and correct or remove it.
- Detect duplicate data, and remove the duplicates.

## 1. Delete Missing Data

### 1.1. Google Play Store

In [9]:
index = -1
for row in android:
    index += 1
    len_header = len(android_header) #length of header
    len_row = len(row)
    if len_row != len_header:
        print("The row has been wrong data or missing value: ",index)
        print(android[index])

The row has been wrong data or missing value:  10472
['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


We have found an error in row 10472th of **googleplaystore.csv** and we can delete or add information for missing value. In this case, i will delete them.

In [10]:
print("Before delete: ",len(android))
del android[10472] #delete row 10472
print("After delete: ",len(android))

Before delete:  10841
After delete:  10840


### 1.2. Apple Store

In [11]:
index = 0
for row in ios:
    index += 1
    len_header = len(ios_header) #length of header
    len_row = len(row)
    if len_row != len_header:
        print("The row has been wrong data or missing value: ",index)
        print(ios[index])

Don't have any wrong or missing data in this dataset 

## 2. Remove Duplicate Apps

shall we check if the dataset has duplicate entries?

### 2.1. Google Play Store

In [12]:
# Create a dictionary of name apps
name_app_Android = {}
for app in android:
    name = app[0] 
    if name not in name_app_Android:
        name_app_Android[name] = 1
    else:
        name_app_Android[name] +=1

In [13]:
name_app_Android

{'Photo Editor & Candy Camera & Grid & ScrapBook': 1,
 'Coloring book moana': 2,
 'U Launcher Lite – FREE Live Cool Themes, Hide Apps': 1,
 'Sketch - Draw & Paint': 1,
 'Pixel Draw - Number Art Coloring Book': 1,
 'Paper flowers instructions': 1,
 'Smoke Effect Photo Maker - Smoke Editor': 1,
 'Infinite Painter': 1,
 'Garden Coloring Book': 1,
 'Kids Paint Free - Drawing Fun': 1,
 'Text on Photo - Fonteee': 1,
 'Name Art Photo Editor - Focus n Filters': 1,
 'Tattoo Name On My Photo Editor': 1,
 'Mandala Coloring Book': 1,
 '3D Color Pixel by Number - Sandbox Art Coloring': 1,
 'Learn To Draw Kawaii Characters': 1,
 'Photo Designer - Write your name with shapes': 1,
 '350 Diy Room Decor Ideas': 1,
 'FlipaClip - Cartoon animation': 1,
 'ibis Paint X': 1,
 'Logo Maker - Small Business': 1,
 "Boys Photo Editor - Six Pack & Men's Suit": 1,
 'Superheroes Wallpapers | 4K Backgrounds': 1,
 'Mcqueen Coloring pages': 2,
 'HD Mickey Minnie Wallpapers': 1,
 'Harley Quinn wallpapers HD': 1,
 'Color

**We have duplicate entries and duplicate counts (more than one)**

But what's so special about it ?

Let's take a look at a duplicate application

In [14]:
for app in android:
    name = app[0]
    if name == "imo free video calls and chat":
        print(app)

['imo free video calls and chat', 'COMMUNICATION', '4.3', '4785892', '11M', '500,000,000+', 'Free', '0', 'Everyone', 'Communication', 'June 8, 2018', '9.8.000000010501', '4.0 and up']
['imo free video calls and chat', 'COMMUNICATION', '4.3', '4785988', '11M', '500,000,000+', 'Free', '0', 'Everyone', 'Communication', 'June 8, 2018', '9.8.000000010501', '4.0 and up']
['imo free video calls and chat', 'COMMUNICATION', '4.3', '4785988', '11M', '500,000,000+', 'Free', '0', 'Everyone', 'Communication', 'June 8, 2018', '9.8.000000010501', '4.0 and up']
['imo free video calls and chat', 'COMMUNICATION', '4.3', '4785892', '11M', '500,000,000+', 'Free', '0', 'Everyone', 'Communication', 'June 8, 2018', '9.8.000000010501', '4.0 and up']


  I noticed that the apps differ only in the number of reviews. This may be data collected at different times. So I will choose the apps with the largest number of reviews  because the higher the number of reviews, the more reliable the ratings.

First, I create a dictionary with the maximum number of reviews

In [15]:
reviews_max = {}
for app in android:
    name = app[0]
    n_reviews = int(app[3])   
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

In [16]:
print("length expect: ",len(reviews_max))
reviews_max

length expect:  9659


{'Photo Editor & Candy Camera & Grid & ScrapBook': 159,
 'Coloring book moana': 974,
 'U Launcher Lite – FREE Live Cool Themes, Hide Apps': 87510,
 'Sketch - Draw & Paint': 215644,
 'Pixel Draw - Number Art Coloring Book': 967,
 'Paper flowers instructions': 167,
 'Smoke Effect Photo Maker - Smoke Editor': 178,
 'Infinite Painter': 36815,
 'Garden Coloring Book': 13791,
 'Kids Paint Free - Drawing Fun': 121,
 'Text on Photo - Fonteee': 13880,
 'Name Art Photo Editor - Focus n Filters': 8788,
 'Tattoo Name On My Photo Editor': 44829,
 'Mandala Coloring Book': 4326,
 '3D Color Pixel by Number - Sandbox Art Coloring': 1518,
 'Learn To Draw Kawaii Characters': 55,
 'Photo Designer - Write your name with shapes': 3632,
 '350 Diy Room Decor Ideas': 27,
 'FlipaClip - Cartoon animation': 194216,
 'ibis Paint X': 224399,
 'Logo Maker - Small Business': 450,
 "Boys Photo Editor - Six Pack & Men's Suit": 654,
 'Superheroes Wallpapers | 4K Backgrounds': 7699,
 'Mcqueen Coloring pages': 65,
 'HD Mi

In [17]:
clean_android = [] # list after cleaning duplicate entries
already_added = [] # list check names that have already been added
for app in android:
    name = app[0]
    n_reviews = int(app[3])
    if (name not in already_added) and (reviews_max[name] == n_reviews):
        clean_android.append(app)
        already_added.append(name)

Explore the **clean_android**:

In [18]:
explore_data(clean_android)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']
['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']
['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']
['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']
['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']
Number of rows: 9659


The new dataset has 9659 rows as expected

### 2.2. Apple Store
Do the same as above

In [19]:
name_app_ios = {}
for app in ios:
    name = app[1] 
    if name not in name_app_ios:
        name_app_ios[name] = 1
    else:
        name_app_ios[name] +=1
print('The apps are duplicated:\n')
for key in name_app_ios:
    if name_app_ios[key] > 1:
        print(key)        

The apps are duplicated:

Mannequin Challenge
VR Roller Coaster


In [20]:
reviews_max = {}
for app in ios:
    name = app[1]
    n_reviews = int(app[5])   
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

In [21]:
print("length expect: ",len(reviews_max))
reviews_max

length expect:  7195


{'Facebook': 2974676,
 'Instagram': 2161558,
 'Clash of Clans': 2130805,
 'Temple Run': 1724546,
 'Pandora - Music & Radio': 1126879,
 'Pinterest': 1061624,
 'Bible': 985920,
 'Candy Crush Saga': 961794,
 'Spotify Music': 878563,
 'Angry Birds': 824451,
 'Subway Surfers': 706110,
 'Fruit Ninja Classic': 698516,
 'Solitaire': 679055,
 'CSR Racing': 677247,
 'Crossy Road - Endless Arcade Hopper': 669079,
 'Injustice: Gods Among Us': 612532,
 'Hay Day': 567344,
 'Clear Vision (17+)': 541693,
 'Minecraft: Pocket Edition': 522012,
 'PAC-MAN': 508808,
 'Calorie Counter & Diet Tracker by MyFitnessPal': 507706,
 'DragonVale': 503230,
 'The Weather Channel: Forecast, Radar & Alerts': 495626,
 'Head Soccer': 481564,
 'Google – Search made just for mobile': 479440,
 'Despicable Me: Minion Rush': 464312,
 'The Sims™ FreePlay': 446880,
 'Google Earth': 446185,
 'Plants vs. Zombies': 426463,
 'Sonic Dash': 418033,
 'Groupon - Deals, Coupons & Discount Shopping App': 417779,
 '8 Ball Pool™': 416736,


In [22]:
clean_ios = [] # list after cleaning duplicate entries
already_added = [] # list check names that have already been added
for app in ios:
    name = app[1]
    n_reviews = int(app[5])
    if (name not in already_added) and (reviews_max[name] == n_reviews):
        clean_ios.append(app)
        already_added.append(name)

## 3. Removing Non-English Apps

I Remove non-English apps because I think non-English apps are not aimed at **Global** users

In [23]:
def non_English(string):
    count_symbol=0
    for char in string:
        if ord(char) >127:
            count_symbol+=1
    if count_symbol >3:
            return True
    return False

### 3.1. Google Play Store

In [24]:
android_english = []
for app in clean_android:
    name = app[0]
    if non_English(name) == False:
        android_english.append(app)

In [25]:
print("Length after removing: ",len(android_english))

Length after removing:  9614


### 3.2. Apple Store

In [26]:
ios_english = []
for app in clean_ios:
    name = app[1]
    if non_English(name) == False:
        ios_english.append(app)

In [27]:
print("Length of data set after removing: ",len(ios_english))

Length of data set after removing:  6181


## 4. Isolating the Free Apps

Remove apps that aren't free because those apps can't approach everyone

### 4.1. Google Playstore

In [28]:
android_free = []
for app in android_english:
    price = app[7]
    if price == '0':
        android_free.append(app)

In [29]:
print("Length of data set after ioslating the free apps: ",len(android_free))

Length of data set after ioslating the free apps:  8864


### 4.2. Apple Store

In [30]:
ios_free = []
for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_free.append(app)

In [31]:
print("Length of data set after ioslating the free apps: ",len(ios_free))

Length of data set after ioslating the free apps:  3220


# IV. Analysis Dataset

## 1. Most Common Apps by Genre

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:
1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful on both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification.

In [32]:
def freq_table (dataset, index):
    table_percentage ={}
    total = 0
    for row in range (0,len(dataset)):
        total +=1
        genre = dataset[row][index]
        if genre in table_percentage:
            table_percentage[genre] += 1
        else:
            table_percentage[genre] = 1
    for key in table_percentage:
        table_percentage[key] = float(table_percentage[key]*100)/total
    return table_percentage

In [33]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

### 1.1. Google Play Store

In [34]:
print("Catagory(%): \n")
display_table(android_free, 1)

Catagory(%): 

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.700361010830325
MEDICAL : 3.5311371841155235
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.237815884476534
HEALTH_AND_FITNESS : 3.079873646209386
PHOTOGRAPHY : 2.9444945848375452
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768953
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418774
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075813
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART

It seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.). However, if we investigate this further, we can see that the family category (which accounts for almost 19% of the apps) means mostly games for kids.

Even so, practical apps seem to have a better representation on Google Play compared to App Store. This picture is also confirmed by the frequency table we see for the Genres column:

In [35]:
print("Genres: \n")
display_table(android_free, -4)

Genres: 

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.700361010830325
Medical : 3.5311371841155235
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.237815884476534
Action : 3.1024368231046933
Health & Fitness : 3.079873646209386
Photography : 2.9444945848375452
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.041967509025271
Dating : 1.861462093862816
Arcade : 1.8501805054151625
Video Players & Editors : 1.7712093862815885
Casual : 1.759927797833935
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418774
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.925090

The difference between the Genres and the Category columns is not crystal clear, but one thing we can notice is that the Genres column is much more granular (it has more categories). We're only looking for the bigger picture at the moment, so we'll only work with the Category column moving forward.

### 1.2. Apple Store

In [36]:
print("Genres: \n")
display_table(ios_free, -5)

Genres: 

Games : 58.13664596273292
Entertainment : 7.888198757763975
Photo & Video : 4.968944099378882
Education : 3.6645962732919255
Social Networking : 3.2919254658385095
Shopping : 2.608695652173913
Utilities : 2.515527950310559
Sports : 2.142857142857143
Music : 2.049689440993789
Health & Fitness : 2.018633540372671
Productivity : 1.7391304347826086
Lifestyle : 1.5838509316770186
News : 1.3354037267080745
Travel : 1.2422360248447204
Finance : 1.1180124223602483
Weather : 0.8695652173913043
Food & Drink : 0.8074534161490683
Reference : 0.5590062111801242
Business : 0.5279503105590062
Book : 0.43478260869565216
Navigation : 0.18633540372670807
Medical : 0.18633540372670807
Catalogs : 0.12422360248447205


We can see that among the free English apps, more than a half (58%) are **games**.  **Entertainment** apps are close to (7%), followed by **photo and video apps**, which are close to (5%). Only (3.6%) of the apps are designed for **education**, followed by **social networking** apps which amount for (3.3%) of the apps in our data set.

The general impression is that App Store (at least the part containing free English apps) is dominated by apps that are designed for fun **(games, entertainment, photo and video, social networking, sports, music, etc.)**, while apps with practical purposes **(education, shopping, utilities, productivity, lifestyle, etc.)** are more rare. 

However, the fact that fun apps are the most numerous doesn't also imply that they also have the greatest number of users — the demand might not be the same as the offer.



**SUMMARY**:
                   
Up to this point, we found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps. Now we'd like to get an idea about the kind of apps that have most users.

## 2. Popular Apps

## 2.1. Most Popular Apps by Genre on the App Store
One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but for the App Store data set this information is missing. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

Below, we calculate the average number of user ratings per app genre on the App Store:

In [37]:
genres_ios = freq_table(ios_free, -5)
for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_free:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22812.92467948718
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


On average, navigation apps have the highest number of user reviews, but this figure is heavily influenced by Waze and Google Maps, which have close to half a million user reviews together:

In [38]:
for app in ios_free:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5]) # print name and number of rating

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


The same pattern applies to social networking apps, where the average number is heavily influenced by a few giants like Facebook, Pinterest, Skype, etc. Same applies to music apps, where a few big players like Pandora, Spotify, and Shazam heavily influence the average number.

Our aim is to find popular genres, but navigation, social networking or music apps might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000 threshold. We could get a better picture by removing these extremely popular apps for each genre and then rework the averages, but we'll leave this level of detail for later.

**Reference** apps have 74,942 user ratings on average, but it's actually the Bible and Dictionary.com which skew up the average rating:

In [39]:
for app in ios_free:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


However, this niche seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

* Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

* Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.

* Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

Now let's analyze the Google Play market a bit.

## 2.2. Most Popular Apps by Genre on Google Play
For the Google Play market, we actually have data about the number of installs, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.):

In [40]:
display_table(android_free, 5) # the Installs columns

1,000,000+ : 15.72653429602888
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.1985559566787
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.772111913357401
5,000+ : 4.512635379061372
10+ : 3.542418772563177
500+ : 3.2490974729241877
50,000,000+ : 2.3014440433212995
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.7897111913357401
1+ : 0.5076714801444043
500,000,000+ : 0.27075812274368233
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


One problem with this data is that is not precise. For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to get an idea which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

To perform computations, however, we'll need to convert each install number to float — this means that we need to remove the commas and the plus characters, otherwise the conversion will fail and raise an error. We'll do this directly in the loop below, where we also compute the average number of installs for each genre (category).

In [41]:
categories_android = freq_table(android_free, 1)
for category in categories_android:
    total = 0
    len_category = 0
    for app in android_free:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [42]:
for app in android_free:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

If we removed all the communication apps that have over 100 million installs, the average would be reduced roughly ten times:

In [43]:
under_100_m = []
for app in android_free:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100_m.append(float(n_installs))
        
sum(under_100_m) / len(under_100_m)

3603485.3884615386


We see the same pattern for the video players category, which is the runner-up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.

The books and reference genre looks fairly popular as well, with an average number of installs of 8,767,811. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

Let's take a look at some of the apps from this genre and their number of installs:

In [44]:
for app in android_free:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E


The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average:

In [45]:
for app in android_free:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


However, it looks like there are only a few very popular apps, so this market still shows potential. Let's try to get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity (between 1,000,000 and 100,000,000 downloads):

In [46]:
for app in android_free:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.

We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

However, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

# V. Conclusions

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

# References 

* [Dataquest.io](https://github.com/dataquestio/solutions/blob/master/Mission350Solutions.ipynb)